tests/rados/test_rados_watcher_cleanup.py

```diff
@@ -0,0 +1,564 @@
+        )
+
+        # --- Phase 2: Delete Watchers based on various methods ---
+        log.info("\n--- Phase 2: Initiating Watcher Cleanup Verification ---")
+        cleanup_methods = ["unmap", "umount", "blacklist", "restart", "crash"]
+        # Iterate through mapped devices to apply cleanup methods
+        for client_hostname, images_info in list(mapped_devices.items()):
+            # Using the imported get_host_obj_from_hostname()
+            client = get_host_obj_from_hostname(
+                hostname=client_hostname, rados_obj=rados_obj
+            )
+
+            if not client:
+                log.warning(
+                    "Client %s not found, skipping cleanup for its images.",
+                    client_hostname,
+                )
+                continue
+            for image_name, device_path in list(images_info.items()):
+                log.info(
+                    "Testing cleanup for Image: %s on Client: %s",
+                    image_name,
+                    client.hostname,
+                )
+                for method in cleanup_methods:
+                    log.info(
+                        "--- Applying cleanup method: '%s' for %s on %s ---",
+                        method,
+                        image_name,
+                        client.hostname,
+                    )
+                    try:
+                        # Re-map/re-mount if necessary for each cleanup method, ensuring state for the test
+                        log.info(
+                            "Re-establishing watcher for %s before '%s' cleanup test.",
+                            image_name,
+                            method,
+                        )
+                        _force_cleanup_device(client, device_path, image_name)
+                        _verify_watcher_cleanup(
+                            rados_obj, pool, image_name, f"pre-test cleanup {method}"
+                        )
+
+                        # Remap and remount for the test
+                        mount_path = "/mnt/%s" % image_name
+                        re_mapped_device, re_mounted_path = (
+                            rados_obj.mount_image_on_client(
+                                pool_name=pool,
+                                img_name=image_name,
+                                client_obj=client,
+                                mount_path=mount_path,
+                            )
+                        )
+                        if not re_mapped_device:
+                            raise Exception(
+                                "Failed to re-map and mount image for cleanup test."
+                            )
+
+                        _verify_watcher_presence(
+                            image_name, client, pool, rados_obj, client.ip_address
+                        )
+
+                        _delete_watcher_and_verify(
```

Comment:
This method does not take care of these workflows correctly. ["restart", "crash", "umount"].. We're passing various workflow names, but doing same operation underneath. 

And why is `_force_cleanup_device()` called everywhere? We need to test various workflows. not just one.